test/v1/domain/StatusReasonEnumSpec.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.domain
+
+import support.UnitSpec
+import utils.enums.EnumJsonSpecSupport
+import v1.models.domain.StatusReasonEnum
+
+class StatusReasonEnumSpec extends UnitSpec with EnumJsonSpecSupport {
+
+  testRoundTrip[StatusReasonEnum](
+    ("signUpReturnAvailable", StatusReasonEnum.signUpReturnAvailable),
+    ("signUpNoReturnAvailable", StatusReasonEnum.signUpNoReturnAvailable),
+    ("itsaFinalDeclaration", StatusReasonEnum.itsaFinalDeclaration),
+    ("itsaQ4Declaration", StatusReasonEnum.itsaQ4Declaration),
+    ("cesaSaReturn", StatusReasonEnum.cesaSaReturn),
+    ("complex", StatusReasonEnum.complex),
+    ("ceasedIncomeSource", StatusReasonEnum.ceasedIncomeSource),
+    ("reinstatedIncomeSource", StatusReasonEnum.reinstatedIncomeSource),
+    ("rollover", StatusReasonEnum.rollover)
+  )
+
+  "toDesViewString" must {
```

Comment:
This test wasn't really needed before, but definitely won't be needed now if they're all vals